app/models/vm_or_template/operations/configuration.rb

```diff
@@ -74,8 +74,10 @@ def disconnect_floppies
   def raw_add_disk(disk_name, disk_size_mb, options = {})
     raise _("VM has no EMS, unable to add disk") unless ext_management_system
     if options[:datastore]
-      datastore = Storage.find_by(:name => options[:datastore])
-      raise _("Data Store does not exist, unable to add disk") unless datastore
+      datastore = ext_management_system.hosts.detect do |h|
+        h.writable_storages.find_by(:name => options[:datastore])
+      end.uniq.compact.first
```

Comment:
This is not going to do what you want.  The `detect` method will return a single `Host` object or `nil` and it looks like you are expecting a `Storage` object array (I think) since you are calling `uniq.compact.first`.

Which brings us to my second point: Test!